drivers/SmartThings/zwave-switch/src/qubino-switches/qubino-relays/qubino-flush-2-relay/init.lua

```diff
@@ -11,43 +11,163 @@
+    return parent
+  else
+    return parent:get_child_by_parent_assigned_key(string.format("%02X", src_channel))
+  end
+end
+
+local function device_init(driver, device)
+  if device.network_type == st_device.NETWORK_TYPE_ZWAVE then
+    device:set_find_child(find_child)
+    device:set_component_to_endpoint_fn(component_to_endpoint)
+  end
+end
+
+local function get_child_metadata(device, endpoint)
+  local name
+  local profile_name
+  if endpoint ~= 3 then
+    name = string.format("Qubino Switch %d", endpoint)
+    profile_name = "metering-switch"
+  else
+    name = "Qubino Temperature Sensor"
+    profile_name = "child-temperature"
+  end
+
+  return {
+    type = "EDGE_CHILD",
+    label = name,
+    profile = profile_name,
+    parent_device_id = device.id,
+    parent_assigned_child_key = string.format("%02X", endpoint),
+    vendor_provided_label = name
+  }
+end
+
+local function device_added(driver, device)
+  if device.network_type ~= st_device.NETWORK_TYPE_CHILD then
+    driver:try_create_device(get_child_metadata(device, 2))
+    if device:is_cc_supported(cc.SENSOR_MULTILEVEL) then
+      driver:try_create_device(get_child_metadata(device, 3))
     end
+  end
+end
+
+local function send_refresh_to_endpoint(ep, driver, device)
+  if ep ~= 3 then
+    device:send(SwitchBinary:Get({}, { dst_channels = { ep } }))
+    device:send(Meter:Get({ scale = Meter.scale.electric_meter.WATTS }, { dst_channels = { ep } }))
+    device:send(Meter:Get({ scale = Meter.scale.electric_meter.KILOWATT_HOURS }, { dst_channels = { ep } }))
+  else
+    device:send(SensorMultilevel:Get({ sensor_type = SensorMultilevel.sensor_type.TEMPERATURE }, { dst_channels = { ep } }))
+  end
 end
 
-local function endpoint_to_component(device, ep)
-    if ep == 2 then
-      return string.format("switch%d", ep)
-    elseif ep == 3 then
-      return "extraTemperatureSensor"
+local function do_refresh(driver, device)
+  for i = 1, 2, 1 do
+    send_refresh_to_endpoint(i , driver, device)
+  end
+  if device:is_cc_supported(cc.SENSOR_MULTILEVEL, 3) then
+    send_refresh_to_endpoint(3 , driver, device)
+  end
+end
+
+local function switch_report_handler(driver, device, cmd)
+  local event
+  local value = cmd.args.target_value or cmd.args.value
+
+  if value ~= nil then
+    if value == SwitchBinary.value.OFF_DISABLE then
+      event = capabilities.switch.switch.off()
     else
-      return "main"
+      event = capabilities.switch.switch.on()
     end
+  end
+
+  device:emit_event_for_endpoint(cmd.src_channel, event)
 end
 
-local function map_components(self, device)
-    device:set_endpoint_to_component_fn(endpoint_to_component)
-    device:set_component_to_endpoint_fn(component_to_endpoint)
+local function set_switch(value)
+  return function(driver, device, cmd)
+    local endpoint = device:component_to_endpoint(cmd.component)
+    local delay = constants.DEFAULT_GET_STATUS_DELAY
+    local query_device = function()
+      for _, ep in pairs(endpoint) do
```

Comment:
this is written as though `component_to_endpoint` will ever return anything other than `{1}`

Reply:
```
function ZwaveChildDevice:component_to_endpoint(component_id)
  local com_to_ep_fn = self:get_field(ZwaveDevice.COMPONENT_TO_ENDPOINT_FUNC)
  if com_to_ep_fn ~= nil then
    return ZwaveDevice.component_to_endpoint(self, component_id)
  else
    return self:get_dst_channel()
  end
end
```
If child device call this function it will return dst_channel `{2}`